openapi_codec/encode.py

```diff
@@ -23,15 +24,61 @@ def generate_swagger_object(document):
         swagger['schemes'] = [parsed_url.scheme]
 
     swagger['paths'] = _get_paths_object(document)
+    swagger['definitions'] = _get_definitions(document)
 
     return swagger
 
 
+def _get_definitions(document):
+
+    definitions = OrderedDict()
+    links = _get_links(document)
+
+    def get_field_def_data(field_item, defs):
+
+        definition_data = {
+            'type': 'object',
+            'properties': {}
+        }
+
+        if isinstance(field_item, coreschema.Object):
+            props = field_item.properties
+        else:
+            props = field_item.schema.properties
+
+        for f_name, f_schema in iter(props.items()):
+
+            if _get_field_type(f_schema) is 'object':
+                defs[f_name] = get_field_def_data(f_schema, defs)
+                definition_data['properties'][f_name] = {
+                    '$ref': '#/definitions/{}'.format(f_name)
+                }
+            else:
+                definition_data['properties'][f_name] = {
+                    'type': _get_field_type(f_schema),
```

Comment:
`type` can also be an `array` here. Now `items` is missing and the swagger-ui (version 2) breaks.

Reply:
@tuky @tuffnatty 
guys, I'm making the changes as per you proposed to do, but could you please help me find an example of the same name for definitions? So that I can write tests and perform changes real quick. Thanks.

Reply:
Say e.g. you have 4 serializers in 2 different files:
```python
class AuthorSerializer(Serializer):
    name = CharField()

class Song(Serializer):
    author = AuthorSerializer()
    co_authors = AuthorSerializer(many=True)
```

```python
class AuthorSerializer(Serializer):
    birthday = DateField()

class Song(Serializer):
    author = AuthorSerializer()
```

The 2 different `AuthorSerializer` should get the same names when generated into CoreAPI format. Then we have a name clash, when reusing them as `'#/definitions/author'` or `'#/definitions/co_authors_item'` in OpenAPI.

Reply:
@tuky, thanks a lot! I wrote a test, reproduced the clash, working on fix.